### PR TITLE
v0.35 release candidate: GroPoint Modbus; Xbee & Digi WiFi fixes; Prep for batch transmission

### PR DESCRIPTION
**Pull request by aufdenkampe**
This PR from `develop` to `master` prepares us for the next release, which includes these new features and improvements:
- #434 
- #439
- #441 
- #445

This PR is required for our upcoming work on:
- #453

This PR could also optimally include:
- #457

@SRGDamia1 I think we want to begin reviewing and preparing this work for our next release. Let's also explore if it makes sense to release the completed work sooner so that we can have a clean `develop` branch for upcoming work (i.e. #453 and #457).

### CODE REVIEW

**Comment by aufdenkampe** [comment]
@SRGDamia1, can we move forward with releasing this completed work soon, so that we can have a clean develop branch for testing upcoming work, including support for new features with the next release of Monitor My Watershed:
- #453

And also this and other pending PRs for new sensors:
- #457

File: src/LoggerModem.cpp [hunk 1]
```diff
@@ -63,6 +64,11 @@ String loggerModem::getModemName(void) {
     return _modemName;
 }
 
+String loggerModem::getModemDevId(void) {
+    return _modemName + F(" Sn ") + _modemSerialNumber + F(" HwVer ") +
+        _modemHwVersion + F(" FwVer ") + _modemFwVersion;
+}
+
```
Comment (SRGDamia1) [inline src/LoggerModem.cpp:71]: These need to be populated for all modems or moved just to the XBee where they are populated.

File: src/LoggerModem.h [hunk 2]
```diff
@@ -340,6 +365,16 @@ class loggerModem {
...
+    /**
+     * @brief Get a detailed printable description of the modem.
+     *
+     * @note These values are polled for and cached in memory till needed
+     *
+     * @return **String** The concatenated name, hardware version, firmware
+     * version, and serial number fo the modem..
+     */
+    String getModemDevId(void);
+
```
Comment (SRGDamia1) [inline src/LoggerModem.h:377]: Implement for all, or move to XBee

File: src/LoggerModem.h [hunk 3]
```diff
@@ -989,6 +1057,48 @@ class loggerModem {
...
+     */
+    String _modemFwVersion;
+    /**
+     * @brief The modem serial number.
+     *
+     * Set in #modemSetup().
+     * Returned as a portion of the #getModemDevId().
+     */
+    String _modemSerialNumber;
+
```
Comment (SRGDamia1) [inline src/LoggerModem.h:1082]: Implement or move.

File: src/modems/DigiXBeeWifi.cpp [hunk 4]
```diff
@@ -254,44 +482,168 @@ bool DigiXBeeWifi::updateModemMetadata(void) {
...
+               loggerModem::_priorModemTemp);
+    } else {
+        MS_DBG(F("Polling for modem chip temperature is disabled"));
+    }
+
+    // Exit command mode
+    MS_DBG(F("Leaving Command Mode after updating modem metadata:"));
     gsmModem.exitCommand();
 
+    ++updateModemMetadata_cnt;
```
Comment (SRGDamia1) [inline src/modems/DigiXBeeWifi.cpp:564]: Why always bump this and restart after 4?

**Pull request merged**